### PR TITLE
rpc tx fee cap check bugfix

### PR DESCRIPTION
**Pull request by hbandura**
before, it would fallback to compare the fee assuming it was a native transaction if the exchange rate couldn't be retrieved (mostly happens in testnets). Now it will properly fail in that case.

added a few unit tests

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit aadfdb9cd3453ec700d83e05cd067ded63d12948
<details><summary>coverage:  44.5% of statements across all listed packages</summary><br>
<pre>
coverage:  54.1% of statements in consensus/istanbul
coverage:  45.0% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  43.3% of statements in consensus/istanbul/backend/internal/db
coverage:  24.1% of statements in consensus/istanbul/backend/internal/enodes
coverage:  22.4% of statements in consensus/istanbul/backend/internal/replica
coverage:  63.3% of statements in consensus/istanbul/core
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  75.3% of statements in consensus/istanbul/uptime
coverage: 100.0% of statements in consensus/istanbul/uptime/store
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: aea7a5970b
</details>

**Review by Anna45W — APPROVED** [review]

File: internal/ethapi/util.go [hunk 1]
```diff
@@ -11,12 +12,43 @@ import (
...
+	weiCap := getWei(cap)
+	feeCurrency, err := cp.GetCurrency(feeCurrencyAddress)
+	if err != nil {
+		return fmt.Errorf("Can't check tx fee cap: %s", err)
+	}
+	if feeCurrency.CmpToCurrency(fee, weiCap, &currency.CELOCurrency) > 0 {
+		feeFloat := float64(fee.Uint64())
+		curr := "celo"
+		if feeCurrencyAddress != nil {
+			feeFloat /= params.Ether
```
[the remark this answers is not in the document]
Comment (hbandura) [reply internal/ethapi/util.go:31]: Yes. See 
https://github.com/celo-org/celo-blockchain/blob/db4a5efa179a9f6963007d13b9978accdf1e6cd4/cmd/utils/flags.go#L414

and

https://github.com/celo-org/celo-blockchain/blob/db4a5efa179a9f6963007d13b9978accdf1e6cd4/eth/config.go#L49

[the remark this answers is not in the document]
Comment (gastonponti) [reply internal/ethapi/util.go:31]: No no, I mean, the `feeFloat`
For some reason we are not dividing by ether the fee if it is Celo (feeCurrencyAddress == nil)
I would expect to have all the fees divided by ether, and not the ones different from celo

[the remark this answers is not in the document]
Comment (hbandura) [reply internal/ethapi/util.go:31]: you are correct

**Review by gastonponti — APPROVED** [review]
LGTM

**Pull request merged**